core/src/main/java/com/graphhopper/routing/util/FerrySpeedCalculator.java

```diff
@@ -54,7 +56,7 @@ public static double minmax(double speed, DecimalEncodedValue avgSpeedEnc) {
 
     @Override
     public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way, IntsRef relationFlags) {
-        if (!way.hasTag("highway") && way.hasTag("route", FERRIES)) {
+        if (isFerry(way)) {
```

Comment:
Since #2849 the speed parsers considered ways tagged with highway as highways, even when there was a `route=ferry` tag. Now they are considered to be ferries again.